Apps/W1/Shopify/app/src/Products/Codeunits/ShpfyProductAPI.Codeunit.al

```diff
@@ -567,4 +569,65 @@ codeunit 30176 "Shpfy Product API"
         foreach JOption in JOptions do
             Options.Add(JsonHelper.GetValueAsText(JOption, 'id'), JsonHelper.GetValueAsText(JOption, 'name'));
     end;
+
+    /// <summary>
+    /// Publish product to selected Shopify Sales Channels
+    /// </summary>
+    /// <param name="ShopifyProduct">Shopify product to be published</param>
+    internal procedure PublishProduct(ShopifyProduct: Record "Shpfy Product")
+    var
+        SalesChannel: Record "Shpfy Sales Channel";
+        GraphQuery: Text;
+        JResponse: JsonToken;
+    begin
+        if ShopifyProduct.Status <> Enum::"Shpfy Product Status"::Active then
+            exit;
+
+        if not GetSalesChannelsToPublishTo(SalesChannel, ShopifyProduct."Shop Code") then
+            exit;
+
+        GraphQuery := CreateProductPublishGraphQuery(ShopifyProduct, SalesChannel);
+
+        JResponse := CommunicationMgt.ExecuteGraphQL(GraphQuery);
+    end;
+
+    local procedure GetSalesChannelsToPublishTo(var SalesChannel: Record "Shpfy Sales Channel"; ShopCode: Code[20]): Boolean
+    var
+        SalesChannelAPI: Codeunit "Shpfy Sales Channel API";
+    begin
+        SalesChannel.SetRange("Shop Code", ShopCode);
+        if SalesChannel.IsEmpty() then
```

Comment:
What if a user creates another sales channel after this initial import? Do they need to go to Sales Channels page and invoke Get Sales Channels? 

Wouldn't it be better to implement some kind of filtering logic based on CREATED_AT? @AndreiPanko

Reply:
@onbuyuka 
This is for case when current customer migrates from 25 to 26.
The table sales channels will be empty, means they "Add product to Shopify" will stop adding sales channel (currently it activates OnlineStore).
To avoid that behavioral breaking change we:
1. if table is empty import all channels. 
2. find one that has handle = online_store and use it.

So this logic is correct. Used only for migrated tenants (or saves user setting step). 

About default - let's look below.